### PR TITLE
Limit autoprovisioned groups to 15

### PR DESCRIPTION
**Pull request by mwielgus**

### CODE REVIEW

File: cluster-autoscaler/main.go [hunk 1]
```diff
@@ -104,11 +104,12 @@ var (
...
-	nodeAutoprovisioningEnabled  = flag.Bool("node-autoprovisioning-enabled", false, "Should CA autoprovision node groups when needed")
+	writeStatusConfigMapFlag         = flag.Bool("write-status-configmap", true, "Should CA write status information to a configmap")
+	maxInactivityTimeFlag            = flag.Duration("max-inactivity", 10*time.Minute, "Maximum time from last recorded autoscaler activity before automatic restart")
+	maxFailingTimeFlag               = flag.Duration("max-failing-time", 15*time.Minute, "Maximum time from last recorded successful autoscaler run before automatic restart")
+	balanceSimilarNodeGroupsFlag     = flag.Bool("balance-similar-node-groups", false, "Detect similar node groups and balance the number of nodes between them")
+	nodeAutoprovisioningEnabled      = flag.Bool("node-autoprovisioning-enabled", false, "Should CA autoprovision node groups when needed")
+	maxAutoprovisionedNodeGroupCount = flag.Int("max-autoprovisioned-node-group-count ", 15, "The maximum number of autoprovisioned groups in the cluster.")
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/main.go]: There's a trailing space in flag name

File: cluster-autoscaler/core/scale_up.go [hunk 2]
```diff
@@ -295,6 +295,17 @@ func executeScaleUp(context *AutoscalingContext, info nodegroupset.ScaleUpInfo)
 }
 
 func addAutoprovisionedCandidates(context *AutoscalingContext, nodeGroups []cloudprovider.NodeGroup, unschedulablePods []*apiv1.Pod) {
+	autoprovisionedNodeGroupCount := 0
+	for _, group := range nodeGroups {
+		if group.Autoprovisioned() {
+			autoprovisionedNodeGroupCount++
+		}
+	}
+	if autoprovisionedNodeGroupCount >= context.AutoscalingOptions.MaxAutoprovisionedNodeGroupCount {
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/core/scale_up.go]: nit: I don't think we use "context.AutoscalingOptions" anywhere else in the code, is it possible to use just "context.MaxAutoprovisionedNodeGroupCount" here?

File: cluster-autoscaler/main.go (hunk 1 above)
Comment (mwielgus) [reply cluster-autoscaler/main.go]: Good catch! Fixed.

File: cluster-autoscaler/core/scale_up.go (hunk 2 above)
Comment (mwielgus) [reply cluster-autoscaler/core/scale_up.go]: Yes, fixed.

**Pull request merged**